lib/find.js

```diff
@@ -25,20 +26,31 @@ function cacheLevel (options) {
     featureCache = new Map()
   }
   if (options && options.preload) {
-    preCache()
+    const featureFileFd = fs.openSync(FEATURE_FILE_PATH, 'r')
+
+    if (featureFileFd < 0) {
+      throw new Error('Failed to open geo.dat file')
+    }
+
+    preCache(featureFileFd)
+
+    fs.closeSync(featureFileFd)
   }
 }
 
 cacheLevel()
 
 /**
  * A function that will load all features into an unexpiring cache
+ *
+ * @param {number} featureFileFd
+ * @returns {void}
  */
-function preCache () {
+function preCache (featureFileFd) {
   // shoutout to github user @magwo for an initial version of this recursive function
   var preloadFeaturesRecursive = function (curTzData, quadPos) {
-    if (curTzData === 'f') {
-      var geoJson = loadFeatures(quadPos)
+    if (curTzData.pos && curTzData.len) {
```

Comment:
Please also update this one too:

```suggestion
    if (curTzData.pos >= 0 && curTzData.len) {
```

Reply:
Oops! I totally forgot to update that. Thanks for catching this. Fixed.